### PR TITLE
Update GH Workflows, README; fix up docker dev image

### PR DESCRIPTION
**Pull request by bonicim**
* Update Dockerfile to use official Volttron docker image on Dockerhub
* Update README
* Standardize code to use Black
* Add precommit hook

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -140,3 +140,33 @@ grep-cmso
 # grep for campus topic sent from central forwarder
 grep-cmlo
 ```
+
+
+# Development
+
+If you plan on extending or developing the "platform_config.yml", "configs/", or the setup scripts in "core/", build the Docker image, "Dockerfile-dev", only once using docker-compose -f docker-compose-dev.yml build --no-cache volttron1. Then start the container using docker-compose -f docker-compose-dev.yml up. When you want to make changes to "platform_config.yml", "configs/", or "core/", simply make the changes and then rerun your container. You do not have to rebuild the image every time you make changes to those aforementioned files and folders because they are mounted into the container. The only time you should rebuild the image is when you make changes to the "volttron" source code since that is not mounted to the container but rather baked into the image during the image build.
```
Comment (shwethanidd) [inline README.md:147]: I thought you removed Dockerfile-Dev. Could you remove references to Dockerfile-Dev here.

File: README.md [hunk 2]
```diff
@@ -140,3 +140,33 @@ grep-cmso
...
+
+To setup your environment for development, do the following:
+
+```shell
+# give execute permissions to all users
+chmod a+x core/*
+
+# build the development image (only have to do one time)
+docker-compose -f docker-compose-dev.yml build --no-cache volttron1
```
Comment (shwethanidd) [inline README.md:156]: Looks like docker-compose-dev.yml is also removed. So these commands are not valid?

**Review by shwethanidd — CHANGES_REQUESTED** [review]
Minor changes requested.

File: .github/workflows/publish_image.yml [hunk 3]
```diff
@@ -1,3 +1,5 @@
+# This workflow is based on example from #  https://github.com/marketplace/actions/build-and-push-docker-images
```
[the remark this answers is not in the document]
Comment (bonicim) [reply .github/workflows/publish_image.yml]: Already done. It's been removed in [commit 66b7e2](https://github.com/bonicim/volttron-tcc-docker-ecolong/commit/66b7e2d78307fdd62c226d0282e8993a021dad59).

File: README.md (hunk 2 above)
Comment (bonicim) [reply README.md:156]: Yes, those commands are no longer needed. Will update the README.

File: README.md (hunk 1 above)
Comment (bonicim) [reply README.md]: Will do.

**Comment by bonicim** [comment]
> Minor changes requested.

Changes made in commit 5f49bf9.

**Review by shwethanidd — APPROVED** [review]

**Pull request merged**